go-controller/pkg/metrics/ovnkube_controller.go

```diff
@@ -619,6 +647,106 @@ func ipsecMetricHandler(table string, model model.Model) {
 	}
 }
 
+// MonitorIPsecTunnelsState will register a labeled metric for the ipsec tunnel status. It will subscribe
+// to xfrm netlink events and update the metric with the tunnel up or down status.
+func MonitorIPsecTunnelsState(stopChan <-chan struct{}, wg *sync.WaitGroup, ovsVsctl ovsClient, ipsec ipsecClient) {
```

Comment:
would it be easier to unit test if you launched the go routine in a seperate func? then you wouldnt need your sleep in your unit test to wait for goroutine to come up.

Reply:
no. what i meant is its hard to unit test funcs when they launch go routines - theres no option to not launch a go routine if you wanted to test the func serially - in-line - which doesnt sound good for a unit test. Increases chances of go routine leak too between tests.